### PR TITLE
Update Versions.props and PublishData.json for 16.9p2 snap 

### PR DESCRIPTION
**Pull request by allisonchou**

### CODE REVIEW

File: eng/config/PublishData.json [hunk 1]
```diff
@@ -207,15 +207,15 @@
...
       "version": "3.9.*",
       "packageFeeds": "default",
-      "channels": [ "dev16.9p1" ],
+      "channels": [ "dev16.9p2" ],
       "vsBranch": "rel/d16.9",
       "vsMajorVersion": 16
     },
     "master-vs-deps": {
       "nugetKind": ["Shipping", "NonShipping"],
       "version": "3.8.*",
```
Comment (dibarbet) [inline eng/config/PublishData.json:216]: Predates this PR but this should be 3.9.* too, right @JoeRobich ?

**Review by JoeRobich — APPROVED** [review]

**Review by dibarbet — COMMENTED** [review]
lgtm, just think we need to fix the version for packages from master-vs-deps

File: eng/config/PublishData.json (hunk 1 above)
Comment (JoeRobich) [reply eng/config/PublishData.json:216]: Good eye. Now I wonder how we use that property since nothing blew up...

Comment (allisonchou) [reply eng/config/PublishData.json]: Updated! Thanks for the catch.

**Review by dibarbet — APPROVED** [review]

File: eng/config/PublishData.json (hunk 1 above)
Comment (dibarbet) [reply eng/config/PublishData.json]: Looking at the code, I actually cannot see it being used anywhere...
I see no references to the branch's version from the $data object anywhere- https://github.com/dotnet/roslyn/blob/d7048e7263cc33c888d3f46e6097ede64085199b/eng/publish-assets.ps1

**Pull request merged**